### PR TITLE
Add one test case for the `disjointdhcps` feature of command `nodeset`

### PR DESCRIPTION
**Pull request by neo954**
Add one cases in this pull request
This pull request is for task #3487
This pull request is for issue #3088

[case 1]
Case Name: nodeset_disjointdhcps_petitboot
description: Verify the `disjointdhcps` feature when petitboot is used for OS loader.

### CODE REVIEW

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 1]
```diff
@@ -347,3 +347,77 @@ check:rc==1
...
+cmd:makedns -n
+check:rc==0
+cmd:chdef testnode1 netboot=petitboot addkcmdline=debug
+check:rc==0
+cmd:mkdef "rhels7.99-ppc64le-install-compute" -u profile=compute provmethod=install osvers=rhels7.99 osarch=ppc64le
+cmd:mkdir -p /install/rhels7.99/ppc64le
+cmd:mkdir -p /install/rhels7.99/ppc64le/ppc/ppc64le
+cmd:echo blah >/install/rhels7.99/ppc64le/ppc/ppc64le/vmlinuz
+cmd:echo blah >/install/rhels7.99/ppc64le/ppc/ppc64le/initrd.img
+cmd:nodeset testnode1 osimage=rhels7.99-ppc64le-install-compute
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: Before we run nodeset, better to output "sharedtftp" and "disjointdhcps"

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 2]
```diff
@@ -347,3 +347,77 @@ check:rc==1
...
+check:rc==0
+cmd:cp -f /etc/hosts /etc/hosts.xcattestbak
+cmd:echo "10.1.1.200 testnode1" >> /etc/hosts
+cmd:makedns -n
+check:rc==0
+cmd:chdef testnode1 netboot=petitboot addkcmdline=debug
+check:rc==0
+cmd:mkdef "rhels7.99-ppc64le-install-compute" -u profile=compute provmethod=install osvers=rhels7.99 osarch=ppc64le
+cmd:mkdir -p /install/rhels7.99/ppc64le
+cmd:mkdir -p /install/rhels7.99/ppc64le/ppc/ppc64le
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: Why we require Line 364?  `mkdir -p /install/rhels7.99/ppc64le/ppc/ppc64le` could cover the full path.
A bit confusing,  in rhels7.3 iso. `ls /install/rhels7.3/ppc64le/ppc/ppc64/`

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 3]
```diff
@@ -347,3 +347,77 @@ check:rc==1
...
+cmd:echo blah >/install/rhels7.99/ppc64le/ppc/ppc64le/vmlinuz
+cmd:echo blah >/install/rhels7.99/ppc64le/ppc/ppc64le/initrd.img
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: Just touch is enough,  maybe "touch /install/rhels7.99/ppc64le/ppc/ppc64le/{vmlinuz,initrd.img}"

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 4]
```diff
@@ -347,3 +347,77 @@ check:rc==1
...
+check:rc==0
+#cmd:makedhcp -q testnode1 | grep ^testnode1:
+#check:rc==0
+cmd:nodeset testnode1 offline
+check:rc==0
+cmd:grep "debug" /tftpboot/petitboot/testnode1
+check:rc!=0
+cmd:chdef -t site sharedtftp=0
+check:rc==0
+cmd:chdef -t site disjointdhcps=0
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: please merge the two `chdef` in one `cmd`

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 5]
```diff
@@ -347,3 +347,77 @@ check:rc==1
...
+check:rc==0
+cmd:xdsh $$SN 'umount /tftpboot'
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: Need to check if /tftpboot is mouted first and then unmount.

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 6]
```diff
@@ -347,3 +347,77 @@ check:rc==1
...
+cmd:xdsh $$SN 'test -f /tftpboot/petitboot/testnode1'
+check:rc!=0
+cmd:nodeset testnode1 offline
+check:rc==0
+# Clean up
+cmd:chdef -t site sharedtftp=1
+check:rc==0
+cmd:chdef -t site disjointdhcps=0
+check:rc==0
+cmd:xdsh $$SN 'mount /tftpboot'
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: Need to check fstab to see if it required to be mounted again.

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 7]
```diff
@@ -347,3 +347,77 @@ check:rc==1
...
+check:rc==0
+cmd:cp -f /etc/hosts.xcattestbak /etc/hosts
+cmd:getent hosts testnode1 | grep testnode1
+check:rc!=0
+cmd:nodeset testnode1 osimage=rhels7.99-ppc64le-install-compute
+check:rc!=0
+cmd:noderm testnode1
+cmd:rmdef -t osimage -o "rhels7.99-ppc64le-install-compute"
+cmd:rm -rf /install/rhels7.99
+end
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: You still miss the case or sub-case for dhcpserver and dynamic range

**Comment by neo954** [comment]
@robin2008, test case revised. Please take a look if you have time.

File: xCAT-test/autotest/testcase/nodeset/cases0 [hunk 8]
```diff
@@ -347,3 +347,86 @@ check:rc==1
...
+check:rc==0
+#cmd:makedhcp -q testnode1 | grep ^testnode1:
+#check:rc==0
+cmd:nodeset testnode1 offline
+check:rc==0
+cmd:grep "debug" /tftpboot/petitboot/testnode1
+check:rc!=0
+cmd:chdef -t site sharedtftp=0 disjointdhcps=0
+check:rc==0
+cmd:xdsh $$SN 'mount | grep /tftpboot && umount /tftpboot'
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/nodeset/cases0]: could we use a different tftpboot for such fake nodeset testing.
```
mkdir -p /tftpboot1/boot/grub2
touch /tftpboot1/boot/grub2/grub2.ppc
```

**Pull request merged**